Fix israel performance

elaav opened the pull request:
refs #498 
<!-- if your contribution is a fix -->

- [x] Tests with a significant number of years to be tested for your calendar.
- [x] Changelog amended with a mention describing your changes. **Note** *Please do NOT change the version number here. It's the project maintainers' duty.*

<!-- Release management

- Commit for the tag:
    - [ ] Edit version in setup.py
    - [ ] Add version in Changelog.md ; trim things
    - [ ] Push & wait for the tests to be green
    - [ ] tag me.
    - [ ] build sdist + wheel packages (``make package``)
- Back to dev commit:
    - [ ] Edit version in setup.py
    - [ ] Add the "master / nothing to see here" in Changelog.md
    - [ ] Push & wait for the tests to be green
- [ ] Merge --ff
- Github stuff
    - [ ] Push tag in Github
    - [ ] Edit release on Github using the changelog.
    - [ ] Delete branch
- [ ] upload release on PyPI using ``twine``
- [ ] (*optional*) Make feeback on the various PR or issues.

 -->

brunobord commented:
Looks great, but I was wondering: I thought that the way the Israel calendar is built would make sure that if a Hebrew holiday was happening twice the same year, we would detect it. Do you think that your implementation won't be broken if this happens? (and... I'm not sure: can this happen, by the way?)

elaav commented:
> Looks great, but I was wondering: I thought that the way the Israel calendar is built would make sure that if a Hebrew holiday was happening twice the same year, we would detect it. Do you think that your implementation won't be broken if this happens? (and... I'm not sure: can this happen, by the way?)

It was built out of Gregorian thinking and that's why the same holiday was twice a year. In the way I implemented it, it takes the holidays by the Hebrew dates and translates them to gregorian dates. So every holiday will appear once. The only concern which can emerge from the way I implemented it is that it will mistakenly take holidays from the year before/after. But I changed the tests to be more precise, so now we have one to one comparing for the years 2017-2020. And in general, it shouldn't happen since we don't have holidays in December and January (or even November and February) so this leap should never happen.
